controllers/cloudinit/controlplane_init.go

```diff
@@ -149,12 +153,28 @@ func NewInitControlPlane(input *ControlPlaneInput) ([]byte, error) {
 	input.WriteFiles = append(input.WriteFiles, input.AdditionalFiles...)
 	input.SentinelFileCommand = sentinelFileCommand
 	input.ControlPlaneEndpointType = "DNS"
+	major, minor, err := extractVersionParts(input.Version)
+	if err != nil {
+		return nil, err
+	}
+	input.Version = generateSnapChannelArgument(major, minor)
+
+	// Get at least dns enabled
+	if input.Addons == nil {
+		input.Addons = []string{"dns"}
+	}
```

Comment:
is the dns addon one that "we always want there"? If so, maybe check if "dns" is in the list?